pennylane/transforms/decompositions/pauli_based_computation.py

```diff
@@ -22,8 +22,96 @@
 def to_ppr(tape):
     r"""A quantum compilation pass that converts Clifford+T gates into Pauli Product Rotation (PPR)
     gates.
-    """
 
+    .. note::
+
+        This transform requires decorating the QNode with :func:`@qml.qjit <pennylane.qjit>`. In
+        addition, the circuits that generated from this pass are currently not executable on any
+        backend. This pass is only for Pauli-based-computation analysis with the ``null.qubit``
+        device and potential future execution when a suitable backend is available.
+
+    Clifford gates are defined as :math:`\exp(-{iP\tfrac{\pi}{4}})`, where :math:`P` is a Pauli word.
+    Non-Clifford gates are defined as :math:`\exp(-{iP\tfrac{\pi}{8}})`.
+
+    For more information on Pauli product measurements and Pauli product rotations, check out the
+    `compilation hub <https://pennylane.ai/compilation/pauli-product-measurement>`__.
+
+    The full list of supported gates and operations are
+    ``qml.H``,
+    ``qml.S``,
+    ``qml.T``,
+    ``qml.X``,
+    ``qml.Y``,
+    ``qml.Z``,
+    ``qml.adjoint(qml.S)``,
+    ``qml.adjoint(qml.T)``,
+    ``qml.CNOT``, and
+    ``qml.measure`` (with PennyLane program capture enabled: :func:`~.capture.enable`).
+
+    Args:
+        fn (QNode): the QNode to apply the pass to
+
+    Returns:
+        :class:`QNode <pennylane.QNode>`
+
+    .. seealso::
+        :func:`~.transforms.commute_ppr`, :func:`~.transforms.merge_ppr_ppm`,
+        :func:`~.transforms.ppr_to_ppm`, :func:`~.transforms.ppm_compilation`,
+        :func:`~.transforms.reduce_t_depth`, :func:`~.transforms.decompose_arbitrary_ppr`
+
+    .. note::
+
+        For better compatibility with other PennyLane functionality, ensure that PennyLane program
+        capture is enabled with :func:`pennylane.capture.enable`.
+
+    **Example**
+
+    The ``to_ppr`` compilation pass can be applied as a dectorator on a QNode:
```

Comment:
Same for the other docstrings as well

```suggestion
    The ``to_ppr`` compilation pass can be applied as a decorator on a QNode:
```